src/components/Swap/SwapActionButton/index.tsx

```diff
@@ -28,8 +28,8 @@ export default function SwapActionButton() {
       (!permit2Enabled && approval.state !== SwapApprovalState.APPROVED) ||
       error !== undefined ||
       (!isWrap && !trade) ||
-      !(inputCurrencyAmount && inputCurrencyBalance) ||
-      inputCurrencyBalance.lessThan(inputCurrencyAmount),
+      !inputCurrencyAmount ||
+      (Boolean(inputCurrencyBalance) && Boolean(inputCurrencyBalance?.lessThan(inputCurrencyAmount))),
```

Comment:
You either need the guard check (`inputCurrencyBalance &&`) _or_ the optional operator (`?.`) - I think you can simplify it like this:

```suggestion
      inputCurrencyBalance?.lessThan(inputCurrencyAmount),
```

Reply:
still need to cast to Boolean so that `isDisabled` isn't undefined